music21/musicxml/xmlToM21.py

```diff
@@ -3387,6 +3387,10 @@ def xmlTechnicalToArticulation(self, mxObj):
                 #     mallet lift, mallet table, martellato, martellato lift,
                 #     muted martellato, pluck lift, and swing.
                 tech.displayText = mxObj.text
+            if tag == 'fret':
+                tech.number = mxObj.text
```

Comment:
this should be `int(mxObj.text)` wrapped in a `try: ... except ValueError: pass` statement as above.

We generally only add MusicXML features when they import and export, so can you add m21ToXml.py support for fret and string?